src/software/sensor_fusion/vision_detection.h

```diff
@@ -7,7 +7,6 @@
 #include "software/sensor_fusion/ball_detection.h"
```

Comment:
Must not have caught this in a previous review, but I'm not totally sure we need this data class? Because each detection type basically had a corresponding filter, it seems like it may be easier to just create those detection types and pass them right to the filter. Furthermore, this class has its own `timestamp` member which is duplicated with each detection (which has its own timestamp). Possibly out of scope of this PR, just want to point it out.

Reply:
If you look at getVisionDetection, there's a lot of camera code that can exist nicely in one place if that function returns VisionDetection. I think I might be able to remove the extra timestamp, though

Reply:
Hmm fair enough. Probably out of scope for this anyway since it works fine and already existed. We can always come back to this later.